### ISSUE TITLE
void linux openjdk8-jre fails to run older minecraft's

### ISSUE THREAD

**Issue by Ahptt** [PolyMC/PolyMC#1533]
### Operating System

Void Linux

### Version of PolyMC

1.4.3

### Version of Qt

6.2.4

### Description of bug

I run void linux and have the openjdk8-jre package to run Minecraft < 1.12.2 versions (I play competitive 1.8.9 pvp lol). However, on void linux, when booting with this version, the game crashes. I had this issue on Debian, but the fix for me was to use the nvidia-openjdk-8-jre package. Unfortunately, this is not packaged in the void repos (although, there is a way to install deb's, but I'm unaware if that would actually work for something like this.)

### Steps to reproduce

1. Run Minecraft 1.8.9 or 1.12.2 (any version that <java17 is required)
2. Have the openjdk8-jre package installed and set the java installation location set (mine is /usr/lib/jvm/java-1.8-openjdk/jre/bin/java)
3. Run and see it crash

### Suspected cause

Not sure exactly what could be causing this, and maybe this would be something to bring up with the package maintainers. Where this was fixed by using the nvidia-openjdk-8-jre in Debian, it could be a package issue?

Here is my minecraft log:
https://gist.github.com/Ahptt/b46eb9858242ad198f8e8ce07e4110ee

Thanks to anyone for reading & who may know of a remedy!

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by slendidev** [comment]
As in the Log, this is a problem with a mod installed:
`Do not report to Forge! Remove FoamFixAPI (or replace with FoamFixAPI-Lawful) and try again. (foamfix-0.6.3-anarchy-1.8.x.jar)`
This issue can be closed due to it not being a launcher issue.

**Comment by Ahptt** [comment]
Hi, I've disabled the mod, and I still crash. I've created a vanilla 1.8.9 and pointed it to the same java path. Here is the output log.

```
PolyMC version: 1.4.3


Launched instance in online mode

authserver.mojang.com resolves to:
    [13.249.195.58, 13.249.195.46, 13.249.195.34, 13.249.195.62]

session.minecraft.net resolves to:
    [100.24.172.95, 34.197.32.8, 35.153.153.200]

textures.minecraft.net resolves to:
    [13.107.237.66, 13.107.238.66, 2620:1ec:4e:1::66, 2620:1ec:4f:1::66]

api.mojang.com resolves to:
    [13.249.205.68, 13.249.205.116, 13.249.205.98, 13.249.205.36]


Minecraft folder is:
/home/ahpt/.local/share/PolyMC/instances/1.8.91/.minecraft


Java path is:
/usr/lib/jvm/java-1.8-openjdk/jre/bin/java


Checking Java version...
Java is version 1.8.0_322, using 64 (amd64) architecture, from Void.


AMD Ryzen 9 5900X 12-Core Processor
NVIDIA Corporation GA102 [GeForce RTX 3080] (rev a1)

Subsystem: NVIDIA Corporation GA102 [GeForce RTX 3080]

Kernel driver in use: nvidia

Main Class:
  net.minecraft.client.main.Main

Native path:
  /home/ahpt/.local/share/PolyMC/instances/1.8.91/natives

Libraries:
  /home/ahpt/.local/share/PolyMC/libraries/net/java/jinput/jinput/2.0.5/jinput-2.0.5.jar
  /home/ahpt/.local/share/PolyMC/libraries/net/java/jutils/jutils/1.0.0/jutils-1.0.0.jar
  /home/ahpt/.local/share/PolyMC/libraries/org/lwjgl/lwjgl/lwjgl/2.9.4-nightly-20150209/lwjgl-2.9.4-nightly-20150209.jar
  /home/ahpt/.local/share/PolyMC/libraries/org/lwjgl/lwjgl/lwjgl_util/2.9.4-nightly-20150209/lwjgl_util-2.9.4-nightly-20150209.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/mojang/netty/1.8.8/netty-1.8.8.jar
  /home/ahpt/.local/share/PolyMC/libraries/oshi-project/oshi-core/1.1/oshi-core-1.1.jar
  /home/ahpt/.local/share/PolyMC/libraries/net/java/dev/jna/jna/3.4.0/jna-3.4.0.jar
  /home/ahpt/.local/share/PolyMC/libraries/net/java/dev/jna/platform/3.4.0/platform-3.4.0.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/ibm/icu/icu4j-core-mojang/51.2/icu4j-core-mojang-51.2.jar
  /home/ahpt/.local/share/PolyMC/libraries/net/sf/jopt-simple/jopt-simple/4.6/jopt-simple-4.6.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/paulscode/codecjorbis/20101023/codecjorbis-20101023.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/paulscode/codecwav/20101023/codecwav-20101023.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/paulscode/libraryjavasound/20101123/libraryjavasound-20101123.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/paulscode/librarylwjglopenal/20100824/librarylwjglopenal-20100824.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/paulscode/soundsystem/20120107/soundsystem-20120107.jar
  /home/ahpt/.local/share/PolyMC/libraries/io/netty/netty-all/4.0.23.Final/netty-all-4.0.23.Final.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/google/guava/guava/17.0/guava-17.0.jar
  /home/ahpt/.local/share/PolyMC/libraries/org/apache/commons/commons-lang3/3.3.2/commons-lang3-3.3.2.jar
  /home/ahpt/.local/share/PolyMC/libraries/commons-io/commons-io/2.4/commons-io-2.4.jar
  /home/ahpt/.local/share/PolyMC/libraries/commons-codec/commons-codec/1.9/commons-codec-1.9.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/google/code/gson/gson/2.2.4/gson-2.2.4.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/mojang/authlib/1.5.21/authlib-1.5.21.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/mojang/realms/1.7.59/realms-1.7.59.jar
  /home/ahpt/.local/share/PolyMC/libraries/org/apache/commons/commons-compress/1.8.1/commons-compress-1.8.1.jar
  /home/ahpt/.local/share/PolyMC/libraries/org/apache/httpcomponents/httpclient/4.3.3/httpclient-4.3.3.jar
  /home/ahpt/.local/share/PolyMC/libraries/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar
  /home/ahpt/.local/share/PolyMC/libraries/org/apache/httpcomponents/httpcore/4.3.2/httpcore-4.3.2.jar
  /home/ahpt/.local/share/PolyMC/libraries/org/apache/logging/log4j/log4j-api/2.0-beta9-fixed/log4j-api-2.0-beta9-fixed.jar
  /home/ahpt/.local/share/PolyMC/libraries/org/apache/logging/log4j/log4j-core/2.0-beta9-fixed/log4j-core-2.0-beta9-fixed.jar
  /home/ahpt/.local/share/PolyMC/libraries/tv/twitch/twitch/6.5/twitch-6.5.jar
  /home/ahpt/.local/share/PolyMC/libraries/com/mojang/minecraft/1.8.9/minecraft-1.8.9-client.jar

Native libraries:
  /home/ahpt/.local/share/PolyMC/libraries/net/java/jinput/jinput-platform/2.0.5/jinput-platform-2.0.5-natives-linux.jar
  /home/ahpt/.local/share/PolyMC/libraries/org/lwjgl/lwjgl/lwjgl-platform/2.9.4-nightly-20150209/lwjgl-platform-2.9.4-nightly-20150209-natives-linux.jar

Params:
  --username  --version 1.8.9 --gameDir /home/ahpt/.local/share/PolyMC/instances/1.8.91/.minecraft --assetsDir /home/ahpt/.local/share/PolyMC/assets --assetIndex 1.8 --uuid  --accessToken  --userProperties  --userType 

Window size: 854 x 480

Java Arguments:
[-Xms512m, -Xmx4096m, -Duser.language=en]


Minecraft process ID: 26826


[07:14:37] [Client thread/INFO]: Setting user: Ahpt
[07:14:37] [Client thread/INFO]: (Session ID is token:<ACCESS TOKEN>:<PROFILE ID>)
[07:14:37] [Client thread/INFO]: LWJGL Version: 2.9.4
---- Minecraft Crash Report ----
// I feel sad now :(
Time: 1/12/23 7:14 AM
Description: Initializing game
java.lang.ExceptionInInitializerError
	at ave.ar(SourceFile:560)
	at ave.am(SourceFile:361)
	at ave.a(SourceFile:310)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.polymc.impl.OneSixLauncher.invokeMain(OneSixLauncher.java:104)
	at org.polymc.impl.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:175)
	at org.polymc.impl.OneSixLauncher.launch(OneSixLauncher.java:185)
	at org.polymc.EntryPoint.listen(EntryPoint.java:144)
	at org.polymc.EntryPoint.main(EntryPoint.java:74)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 0
	at org.lwjgl.opengl.LinuxDisplay.getAvailableDisplayModes(LinuxDisplay.java:951)
	at org.lwjgl.opengl.LinuxDisplay.init(LinuxDisplay.java:738)
	at org.lwjgl.opengl.Display.<clinit>(Display.java:138)
	... 13 more
A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------
-- Head --
Stacktrace:
	at ave.ar(SourceFile:560)
	at ave.am(SourceFile:361)
-- Initialization --
Details:
Stacktrace:
	at ave.a(SourceFile:310)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.polymc.impl.OneSixLauncher.invokeMain(OneSixLauncher.java:104)
	at org.polymc.impl.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:175)
	at org.polymc.impl.OneSixLauncher.launch(OneSixLauncher.java:185)
	at org.polymc.EntryPoint.listen(EntryPoint.java:144)
	at org.polymc.EntryPoint.main(EntryPoint.java:74)
-- System Details --
Details:
	Minecraft Version: 1.8.9
	Operating System: Linux (amd64) version 6.1.3_1
	Java Version: 1.8.0_322, Void
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Void
	Memory: 574236416 bytes (547 MB) / 649592832 bytes (619 MB) up to 3817865216 bytes (3641 MB)
	JVM Flags: 2 total; -Xms512m -Xmx4096m
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	Launched Version: 1.8.9
	LWJGL: 2.9.4
	OpenGL: ~~ERROR~~ RuntimeException: No OpenGL context found in the current thread.
	GL Caps: 
	Using VBOs: No
	Is Modded: Probably not. Jar signature remains and client brand is untouched.
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: ~~ERROR~~ NullPointerException: null
	Profiler Position: N/A (disabled)
	CPU: <unknown>
#@!@# Game crashed! Crash report saved to: #@!@# /home/ahpt/.local/share/PolyMC/instances/1.8.91/.minecraft/crash-reports/crash-2023-01-12_07.14.37-client.txt
[07:14:37] [Client Shutdown Thread/INFO]: Stopping server
Process exited with code 255.
```

**Comment by Ahptt** [comment]
After reading the log, I found the error `at org.lwjgl.opengl.LinuxDisplay.getAvailableDisplayModes(LinuxDisplay.java:951)`. I did a basic search and found [this article](https://stackoverflow.com/questions/58939641/lwjgl-crashing-at-startup). I didn't have a use case for xrandr on my void linux installation, but per that article; 

> LWJGL uses xrandr under the hood to get the screen resolutions by parsing xrandr -q command output.

I installed xrandr and the versions loaded with no issues.
